bundles/framework/publisher2/tools/SwipeTool.js

```diff
@@ -0,0 +1,60 @@
+import { AbstractPublisherTool } from './AbstractPublisherTool';
+import { SwipeToolhandler } from '../handler/SwipeToolHandler';
+import { SwipeToolComponent } from '../view/swipe/SwipeToolComponent';
+
+export const SWIPE_ID = 'Oskari.mapframework.bundle.layerswipe.plugin.LayerSwipePlugin';
+
+class SwipeTool extends AbstractPublisherTool {
+    constructor (...args) {
+        super(...args);
+        this.index = 5;
+        this.group = 'additional';
+        this.handler = new SwipeToolhandler(this);
+    }
+    getTool () {
+        return {
+            id: SWIPE_ID,
+            title: Oskari.getMsg('Publisher2', 'BasicView.maptools.swipe.label'),
+            config: this.state.pluginConfig || {}
+        };
+    }
+    getComponent () {
+        return {
+            component: SwipeToolComponent,
+            handler: this.handler
+        };
+    }
+    init (data) {
+        super.init(data);
+        // restore state to handler -> passing init data to it
+        this.handler.init(this.getTool().config);
+    }
+    getValues () {
+        if (!this.isEnabled()) {
+            return null;
+        }
+        const pluginConfig = this.getPlugin().getConfig();
+        const value = {
+            configuration: {
+                mapfull: {
```

Comment:
This should have configuration: { layerswipe: { /* something */ } } instead of mapfull. It should work like maplegend and similar where it's not a mapfull plugin that we are controlling but the whole layerswipe bundle.